### PR TITLE
Created Python Contour code to look at temperature variations and changes in NH3 and H20 

### PR DESCRIPTION
**Pull request by mayaafields**
need help improving for loop structure to increase efficiency

### CODE REVIEW

File: python_codes/contour_plots.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+'''Deviations of Temperature and Concentrations'''
+
+'''temperature'''
+
+t_0 = np.zeros((len(temp[0,0,:,0])))
+
+for i in range(len(temp[0,0,:,0])): 
+    t_0 = temp[0,0,i,0]
+print(t_0)
```
Comment (luminoctum) [inline python_codes/contour_plots.py]: This statement can be vectorized:
```
t_0 = temp[0,:,0,0]
```

File: python_codes/contour_plots.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+temp_variations = np.zeros((len(temp[:,0,0,0]),len(temp[0,:,0,0]),len(temp[0,0,:,0])))
+for i in range(len(temp[:,0,0,0])):
+    for j in range(len(temp[0,:,0,0])):
+        for k in range(len(temp[0,0,:,0])):
+            temp_variations[i,j,k] = temp[i,j,k,0] - t_0
```
Comment (luminoctum) [inline python_codes/contour_plots.py]: These can be vectorized as well:
```
ntime, nx1, nx2, nx3 = temp.shape
temp_variations = temp - t_0.reshape((1,nx1,1,1))
```

File: python_codes/contour_plots.py (hunk 1 above)
Comment (luminoctum) [reply python_codes/contour_plots.py]: The first dimension is time, then x1, x2 and x3

File: python_codes/contour_plots.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+print(np.mean(temp_variations))
+temp_small_changes = temp_variations - np.mean(temp_variations)
+print(temp_small_changes)
+
+'''NH3 Variations'''
+mean_NH3 = np.mean(NH3_total[:,:,0])
+NH3_variations = np.zeros((len(NH3_total[:,0,0]),len(NH3_total[0,:,0])))
+for i in range(len(NH3_total[:,0,0])):
+    for j in range(len(NH3_total[0,:,0])):
+            NH3_variations[i,j] = NH3_total[i,j,0] - mean_NH3
```
Comment (luminoctum) [inline python_codes/contour_plots.py]: Do the same thing for these statements

File: python_codes/contour_plots.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+'''H2O Variations'''
+mean_H2O = np.mean(H2O_total[:,:,0])
+H2O_variations = np.zeros((len(H2O_total[:,0,0]),len(H2O_total[0,:,0])))
+for i in range(len(H2O_total[:,0,0])):
+    for j in range(len(H2O_total[0,:,0])):
+            H2O_variations[i,j] = H2O_total[i,j,0] - mean_H2O
```
Comment (luminoctum) [inline python_codes/contour_plots.py]: change this as well

**Review by luminoctum — CHANGES_REQUESTED** [review]
Vectorizing a statement can speed up your code.

File: python_codes/contour_plots.py (hunk 2 above)
Comment (mayaafields) [reply python_codes/contour_plots.py]: what does this mean

Comment (luminoctum) [reply python_codes/contour_plots.py]: It is called [broadcasting](https://www.geeksforgeeks.org/python-broadcasting-with-numpy-arrays/).

File: python_codes/contour_plots.py [hunk 5]
```diff
@@ -1,39 +1,24 @@
 #! /usr/bin/env python3
 from pylab import *
 from netCDF4 import Dataset
-import numpy as np
 
-'''Retrieving Data from the nc files'''
+# Retrieving Data from the nc files
 data = Dataset('/home/mafields/CylconeDiagnostics/data/jupiter2d-long-main.nc')
```
Comment (luminoctum) [inline python_codes/contour_plots.py]: You can use the following statements to separate the file name and the read action.
```
fname = '/home/mafields/CylconeDiagnostics/data/jupiter2d-long-main.nc'
data = Dataset(fname, 'r')
```
'r' standards for "read mode"

**Review by luminoctum — APPROVED** [review]
Good to go. @mayaafields

**Comment by luminoctum** [comment]
Good python code.

**Comment by luminoctum** [comment]
Good to go

**Pull request merged**